OMCompiler/Compiler/NFFrontEnd/NFExpandableConnectors.mo

```diff
@@ -343,7 +399,7 @@ algorithm
     InstNode.updateComponent(comp, node);
 
     // Also mark the component's children as present.
-    if Type.isComplex(Component.getType(comp)) then
+    if Type.isComplex(Component.getType(comp)) or Type.isArray(Component.getType(comp)) then
```

Comment:
I'm not sure what the purpose of this is, is it to also handle arrays of e.g. records? If so it should be `if Type.isComplex(Type.arrayElementType(Component.getType(comp))) then` instead.

Reply:
I add this because if you have an array inside expandale connector it add this array into variables with potentialyPresented. And when this array is flattening all this variables is potentialyPresented. Without this condion flatten variables remain potentialyPresented and filtered later.